### PR TITLE
fix(pwsh): Avoid polluting the global function namespace

### PR DESCRIPTION
**Pull request by segevfiner**
#### Description
This is an alternative to #3386 using `New-Module` to create a new dynamic script module and `Export-ModuleMemeber` to make it not export anything, other than the global prompt function, instead of prefixing the functions.

> #3322 changed the PowerShell init in order to pass the current starship arguments to starship prompt --continuation. This PR reverts these changes and just uses plain &starship prompt --continuation (Invoke-Native isn't really needed there, and the zsh init does not pass along the arguments either) to avoid polluting the global function namespace with Invoke-Native and Get-Arguments. I also moved the continuation prompt code to ensure $STARSHIP_SHELL and $STARSHIP_SESSION_KEY are set beforehand.
> 
> To avoid putting functions into the user scope with common names, this PR prefixes all of these functions with __Starship-. Get-Arguments is removed because it isn't needed for the continuation prompt.

cc @davidkna 

#### Motivation and Context

See above.
Closes #3386

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [x] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by segevfiner** [comment]
Curse you flaky AWS test! 🙃 similar to #3377.

File: src/init/starship.ps1 [hunk 1]
```diff
@@ -1,153 +1,157 @@
...
-            # NOTE: Attempt to trim any "provider prefix" from the path string.
-            # E.g. `Microsoft.PowerShell.Core\FileSystem::Dev:\` -> `Dev:\`
-            LogicalPath =
-                if ($cwd.Path.StartsWith($provider_prefix)) {
-                    $cwd.Path.Substring($provider_prefix.Length)
-                } else {
-                    $cwd.Path
-                };
+$null = New-Module starship {
+    function Get-Arguments {
```
Comment (davidkna) [inline src/init/starship.ps1:4]: Please inline `Get-Arguments` again (`--continuation` doesn't need it after all, and it makes the whole thing a bit more readable).

File: src/init/starship.ps1 [hunk 2]
```diff
@@ -1,153 +1,157 @@
...
-            # Resolve the provider-logical path 
-            # NOTE: Attempt to trim any "provider prefix" from the path string.
-            # E.g. `Microsoft.PowerShell.Core\FileSystem::Dev:\` -> `Dev:\`
-            LogicalPath =
-                if ($cwd.Path.StartsWith($provider_prefix)) {
-                    $cwd.Path.Substring($provider_prefix.Length)
-                } else {
-                    $cwd.Path
-                };
+$null = New-Module starship {
```
Comment (davidkna) [inline src/init/starship.ps1:3]: Please add a comment explaining this line.

File: src/init/starship.ps1 [hunk 3]
```diff
@@ -1,153 +1,157 @@
...
+    function Get-Arguments {
+        function Get-Cwd {
```
Comment (davidkna) [inline src/init/starship.ps1:5]: With the module architecture, I think this should move to the outer scope.

**Review by davidkna — COMMENTED** [review]
I made a few suggestions, but this seems to work well overall.

**Comment by segevfiner** [comment]
@davidkna Changes applied

File: src/init/starship.ps1 [hunk 4]
```diff
@@ -21,133 +23,130 @@ function Get-Arguments {
...
-# Get arguments for starship continuation prompt
-$arguments = Get-Arguments
-$arguments += "--continuation"
+    # Invoke Starship and set continuation prompt
+    Set-PSReadLineOption -ContinuationPrompt (
+        Invoke-Native -Executable ::STARSHIP:: -Arguments @(
+            "prompt",
+            "--continuation"
+        )
+    )
```
Comment (davidkna) [inline src/init/starship.ps1:141]: Please move this after `STARSHIP_SHELL` and `STARSHIP_SESSION_KEY`, or the continuation prompt could get escaped for a different shell.

**Review by davidkna — APPROVED** [review]
LGTM

**Comment by segevfiner** [comment]
Yay... a release was made without this. 🙃

**Comment by davidkna** [comment]
Thanks for figuring out PowerShell modules @segevfiner!

**Pull request merged**